### PR TITLE
fix(tm2/std)!: use snake_case JSON fields for MemFile and MemPackage

### PR DESCRIPTION
**Pull request by thehowl**
This PR changes the MemFile and MemPackage to use `snake_case` fields for their JSON key names. This matches what all other objects in transactions do. A new test ensures that no transaction objects within the VM keeper have uppercase runes within their exported json fields.

**BREAKING CHANGE:** old `m_addpkg` and `m_run` transactions will stop working. Clients will need to update accordingly, and this change needs to be synced like #1939.

cc/ @zivkovicmilos @gnolang/berty @gnolang/onbloc

### CODE REVIEW

File: gno.land/pkg/sdk/vm/package_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"reflect"
+	"strings"
+	"testing"
+	"unicode"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestJSONSnakeCase(t *testing.T) {
```
Comment (zivkovicmilos) [inline gno.land/pkg/sdk/vm/package_test.go:12]: We can run this in parallel

**Review by zivkovicmilos — APPROVED** [review]
8pzaxm

**Comment by zivkovicmilos** [comment]
Let me coordinate a change on the JS clients before we merge it out 🫡

**Comment by zivkovicmilos** [comment]
@thehowl can you merge in the master changes?

**Comment by thehowl** [comment]
@zivkovicmilos done!

**Comment by zivkovicmilos** [comment]
I opened up an analogue PR for the JS clients:
https://github.com/gnolang/gno-js-client/pull/132

**Comment by jinoosss** [comment]
I opened a PR for Adena to support the changes.
https://github.com/onbloc/adena-wallet/pull/522

**Comment by zivkovicmilos** [comment]
For historical record: 

We released `gno-js-client` `v1.3.0` which houses these changes, so this is the earliest version to use with the `master` branch onwards

**Pull request merged**